test/array_of_similar_arrays.jl

```diff
@@ -198,7 +236,61 @@ using UnsafeArrays
         @test empty(A) == empty(B)
     end
 
+    @testset "stats" begin
+        a1 = rand(1,5); a2 = rand(1,5); a3 = rand(1,5)
+        mu_a1 = mean(a1); mu_a2 = mean(a2); mu_a3 = mean(a3)
+
+        VA = VectorOfSimilarArrays([a1, a2, a3])
+        v1 = [1,2,3,4]
+        v2 = v1.*2
+        v2 = v2.+1
+        VV = VectorOfSimilarVectors([v1,v2])
+
+        @testset "sum" begin
+            VA_sum = @inferred(sum(VA))
+            for i in 1:length(VA[1]) 
+                @test @inferred(VA_sum[i]) == a1[i] + a2[i] + a3[i]
+            end
+        end
+
+        @testset "mean" begin
+            VA_mean = @inferred(mean(VA))
+            for i in 1:length(VA[1])
+                diff = @inferred(VA_mean[i]) - @inferred((a1[i]+a2[i]+a3[i])/3)
+                @test isless(diff, eps(Float64)) 
+            end
+        end
+
+        @testset "var" begin
+            VA_var = @inferred(var(VA))
+            for i in 1:length(VA[1])
+                diff = @inferred(var([a1[i], a2[i], a3[i]])) - VA_var[i]
+                @test @inferred(isless(diff, eps(Float64)))
+            end
+        end
+
+        @testset "cov" begin
+            VV_cov = @inferred(cov(VV))
+            VV_var = @inferred(var(VV))
+            diff = VV_cov[1] + VV_cov[6] + VV_cov[11] + VV_cov[16] - sum(VV_var)
+            @test @inferred(isless(diff, eps(Float64)))
+            @test VV_cov == VV_cov'
+        end
+        
+        @testset "cor" begin
+            VV_cor = @inferred(cor(VV))
+            diff = sum(VV_cor - (zeros(size(VV_cor)).+1))
+            @test VV_cor' == VV_cor
+            @test @inferred(isless(diff, eps(Float64)))
+        end
 
+        a1 = a1 .- mu_a1; a2 = a2 .- mu_a2; a3 = a3 .- mu_a3
+        @testset "centered" begin
+            @test isapprox(mean(a1), 0, atol=eps(Float64))
+            @test isapprox(mean(a2), 0, atol=eps(Float64))
+            @test isapprox(mean(a3), 0, atol=eps(Float64))
+        end
+    end
```

Comment:
Does this include the calls to Statistics.cov, etc.? The ones that aren't using e.g. StatsBase.Weights and are thus not testing statsbase_support.jl? Specfically this part of the tests.

Reply:
Yes, the tests without weights should be in here.